tools/mo/openvino/tools/mo/utils/utils.py

```diff
@@ -15,6 +15,36 @@
     import openvino_telemetry as tm
 except ImportError:
     import openvino.tools.mo.utils.telemetry_stub as tm
+import logging as log
+
+def singleton(class_):
+    instances = {}
+    def getinstance(*args, **kwargs):
+        if class_ not in instances:
+            instances[class_] = class_(*args, **kwargs)
+        return instances[class_]
+    return getinstance
+
+@singleton
+class TempFilesCleaner(object):
```

Comment:
Why is it a singleton?

Reply:
This class is implemented as global garbage collector - you may call it from any place, and it will clean files on demand (by calling `clean()`), or after scope is finished (`with` semantic, even if exception happened) or at process finished (inside destructor, useful when function is called directly from any other place instead of `main()`)

Reply:
I decided to simplify general logic:
1. Instead of `with` I decided just to call `del ngraph_function` and `TempFilesCleaner().clean()` to make it more logically connected
2. `__del__` is removed because `It is not guaranteed that __del__() methods are called for objects that still exist when the interpreter exits.` - https://docs.python.org/2/reference/datamodel.html#object.__del__

Reply:
Deleted at all